include/RAJA/policy/cuda/forall.hpp

```diff
@@ -216,6 +218,76 @@ RAJA_INLINE void forall_impl(cuda_exec<BlockSize, Async>,
     RAJA_FT_END;
   }
 }
+template <typename Iterable, typename LoopBody, size_t BlockSize, bool Async>
+RAJA_INLINE RAJA::resources::Event forall_impl(RAJA::resources::Resource *res,
+                                               cuda_exec<BlockSize, Async>,
+                                               Iterable&& iter,
+                                               LoopBody&& loop_body)
+{
+  using Iterator  = camp::decay<decltype(std::begin(iter))>;
+  using LOOP_BODY = camp::decay<LoopBody>;
+  using IndexType = camp::decay<decltype(std::distance(std::begin(iter), std::end(iter)))>;
+
+  auto func = impl::forall_cuda_kernel<BlockSize, Iterator, LOOP_BODY, IndexType>;
+
+  RAJA::resources::Cuda cuda_res;
+  cudaStream_t stream;
+  if (res){
```

Comment:
Is it better to error out if the user passes in a non-CUDA resource with a CUDA exec policy?